### DISCUSSION TITLE
Manage multiple toolchain versions in parallel with home-manager?

### DISCUSSION THREAD

**Discussion by sebasmagri** [nix-community/fenix#98] [Q&A] [answered]
When you to have stable and nightly available for different projects that need different toolchain versions, how do you manage that?

I'm using home-manager preferably so I'd like to have a way to install multiple toolchains in parallel without collisions. I was looking into having the binaries wrapped like in `cargo-nightly`/`rustfmt-nightly` fashion, but I don't find a way to still avoid the collisions.

**Comment by figsoda** [comment, marked as answer]
The preferred way to do this would probably be to have separate development shells for different projects. A workaround you can do is creating wrappers for different toolchains, something like this:
```nix
writeShellScriptBin "cargo-nightly" ''
  PATH=${fenix.complete.toolchain}/bin:$PATH cargo "$@"
''
```
